test/instances/lseth/LsETHHyperdrive.t.sol

```diff
@@ -0,0 +1,901 @@
+        );
+        uint256 hyperdriveSharesBefore = RIVER.balanceOf(address(hyperdrive));
+        uint256 sharesPaid = RIVER.sharesFromUnderlyingBalance(basePaid);
+        RIVER.approve(address(hyperdrive), sharesPaid);
+        openLong(bob, sharesPaid, false);
+        assertEq(
+            RIVER.balanceOf(address(hyperdrive)),
+            hyperdriveSharesBefore + sharesPaid
+        );
+    }
+
+    // /// Long ///
+
+    function test_open_long_with_eth(uint256 basePaid) external {
+        // Bob opens a long by depositing ETH. This is not allowed and
+        // should throw an unsupported token exception.
+        vm.startPrank(bob);
+        basePaid = basePaid.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxLong(hyperdrive)
+        );
+        vm.expectRevert(IHyperdrive.UnsupportedToken.selector);
+        hyperdrive.openLong{ value: basePaid }(
+            basePaid,
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+    }
+
+    function test_open_long_with_lseth(uint256 basePaid) external {
+        // Get some balance information before the deposit.
+        uint256 totalSharesBefore = RIVER.totalSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Calculate the maximum amount of basePaid we can test. The limit is
+        // either the max long that Hyperdrive can open or the amount of lsETH
+        // tokens the trader has.
+        uint256 maxLongAmount = HyperdriveUtils.calculateMaxLong(hyperdrive);
+        uint256 maxEthAmount = RIVER.underlyingBalanceFromShares(
+            RIVER.balanceOf(bob)
+        );
+
+        // Bob opens a long by depositing lsETH.
+        vm.startPrank(bob);
+        basePaid = basePaid.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            maxLongAmount > maxEthAmount ? maxEthAmount : maxLongAmount
+        );
+        uint256 sharesPaid = RIVER.sharesFromUnderlyingBalance(basePaid);
+        RIVER.approve(address(hyperdrive), sharesPaid);
+        openLong(bob, sharesPaid, false);
+
+        // Ensure that River aggregates and the token balances were updated
+        // correctly during the trade.
+        verifyDeposit(
+            bob,
+            sharesPaid,
+            false,
+            totalSharesBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    function test_open_long_refunds() external {
+        vm.startPrank(bob);
+
+        // Ensure that the refund fails when Bob sends excess ETH
+        // when opening a long with "asBase" set to true.
```

Comment:
Nit: It's not really the refund that is failing. This will fail before a refund is attempted.